### DISCUSSION TITLE
Merge Patches on master but Minors/Majors on another branch

### DISCUSSION THREAD

**Discussion by yannispgs** [renovatebot/renovate#22432] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### If you're self-hosting Renovate, select which platform you are using.

None

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I would like that :
- patches are merged on the `master` branch (PRs have `master` as base branch)
- minors & majors are merged on the `custom` branch (PRs have `custom` as base branch)

### What you tried so far.

I created 2 `packageRules` based on [group presets](https://docs.renovatebot.com/presets-group/#groupallnonmajor) :
- 1 that targets the patches
- 1 that targets minors/majors

However, I found no way to update the base branch for the PRs created for each group. They always have `master` (the default branch of my repo) even if I try to set the `baseBranches` field to `["custom"]` (the documented behaviour does not cover my use case, but I tried it anyway).

As I find no forum post about this use case, maybe I am missing a Renovate key concept. Is it doable ? 

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by rarkins** [comment]
Include the config you tried already

**Reply by yannispgs** [reply]
Thanks for the quick answer ! 

Here is the config I tried first :
```
{
  "extends": [
    "config:base"
  ],
  "packageRules": [
    {
      "groupName": "Major & minor upgrades on custom",
      "groupSlug": "major-minor-upgrade",
      "matchPackagePatterns": [
        "*"
      ],
      "matchUpdateTypes": [
        "major",
        "minor"
      ],
      "baseBranches": [
        "custom"
      ]
    },
    {
      "groupName": "Patch on master",
      "groupSlug": "patch-upgrade",
      "matchPackagePatterns": [
        "*"
      ],
      "matchUpdateTypes": [
        "patch"
      ],
      "baseBranches": [
        "master"
      ]
    }
  ]
}

```

EDIT: and also tried without the `config:base` 'extends'. Same result: both leads to PRs with `master` as baseBranch

**Reply by rarkins** [reply]
`baseBranches` must be at the top level. If you want both major and custom baseBranches then list both. If you want to match on certain baseBranches within packageRules, then use matchBaseBranches with one of them

**Reply by yannispgs** [reply]
Alright. However, the "matchBaseBranches" only applies a filter for the current `packageRules` block. It does not update Renovate base branch depending on the update type (minor, major, patch) as I want to do.

I don't see any field in the doc that allows updating that PR property: does it exist ?

I just want to know if i'm missing a Renovate feature. If it is not possible, then I will do it with another tool.

**Reply by rarkins** [reply]
I don't understand what you're describing

**Reply by yannispgs** [reply]
I would like to create the following rule: 
- **major** upgrade = new PR from branch `renovate/<upgrade_name>` (head branch) to merge into branch `custom` (base branch)
- **minor** upgrade = new PR from branch `renovate/<upgrade_name>` (head branch) to merge into branch `custom` (base branch)
- **patch** upgrade = new PR from branch `renovate/<upgrade_name>` (head branch) to merge into branch `master` (base branch)

But until now, I never managed to tell Renovate to create PRs with `custom` branch as base branch: Renovate always selects `master` as base branch.

**Reply by viceice** [reply, marked as answer]
- first configure baseBranches, add both branches
- second add a rule with match baseBranches master and update types major/minor and set enabled to false
- add another rule with same base branch and set separate minor/ patch to true

**Reply by viceice** [reply]
remember, renovate will always use the base branch to search for deps. renovate can't create a branch from one base branch and the target another branch to merge to.

**Reply by yannispgs** [reply]
It is working the way I wanted. Thank you for your (very) quick help ! 😄
